modules/340-monitoring-kubernetes/images/oom-kills-exporter/src/main.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2024 Flant JSC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"errors"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/node-problem-detector/pkg/systemlogmonitor/logwatchers/kmsg"
+	"k8s.io/node-problem-detector/pkg/systemlogmonitor/logwatchers/types"
+)
+
+var (
+	local = prometheus.NewRegistry()
+)
+
+func checkMetricExistenceByLabels(metricName string, labels map[string]string, r *prometheus.Registry) bool {
+	mfs, err := r.Gather()
+	if err != nil {
+		log.Println("Error gathering metrics:", err)
+		return false
+	}
+
+	for _, mf := range mfs {
+		if mf.GetName() == metricName {
+			for _, metric := range mf.Metric {
+				labelMap := make(map[string]string)
+				for _, label := range metric.Label {
+					labelMap[*label.Name] = *label.Value
+				}
+
+				match := true
+				for key, value := range labels {
+					if labelMap[key] != value {
+						match = false
+						break
+					}
+				}
+
+				if match {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+func getContainerIDFromLog(line string) (string, error) {
+	var taskMemcg string
+	if matches := regexp.MustCompile(`^oom-kill:(.+)`).FindStringSubmatch(line); matches != nil {
+		log.Print(line)
```

Comment:
debug line?

Reply:
This is the main log to understand what the exporter notices in dmsg. Without logging you get a black box with unclear state if you need information about its operation.